runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -947,17 +947,17 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	 * Checks that the dataAddr field of the indexable object is correct.
 	 * this method is supposed to be called only if offheap is enabled.
 	 *
-	 * @param arrayPtr      Pointer to the indexable object
-	 * @param isValidDataAddrForOffHeapObject	Boolean to determine whether the given indexable object is off heap
-	 * @return if the dataAddr field of the indexable object is correct
+	 * @param arrayPtr[in]						Pointer to the indexable object
+	 * @param isDataAddrForOffHeapObject[out]	set true if the given indexable object is off heap
+	 * @return if the dataAddr field of the indexable object is correct((not heap object case), return false if indexable object is off heap
 	 */
 	MMINLINE bool
-	isValidDataAddr(J9IndexableObject *arrayPtr, bool isValidDataAddrForOffHeapObject)
+	isValidDataAddr(J9IndexableObject *arrayPtr, bool *isDataAddrForOffHeapObject)
```

Comment:
perhaps these 2 methods should be named more specifically: isValidDataAddrForHeapObject 

@dmitripivkine opinion?

Reply:
or isValidDataAddrForAdjacentData

Reply:
latter one is better

Reply:
ok, then to be consistent we should rename the output parameter to be isDataNonAdjacent (instead of isOffheapObject as I suggest in an earlier comment)